refactor(Aged Debtor Report)

lomamech opened the pull request:
- Update the new client infrastructure for Aged Debtor Client
- Use the language service for the translation of all report
- feat(bhReportPeriodSelect) implement the componant who allow to select
  fiscal year Period
- Implement test E2E for Aged Debtor Report

closes #1788

Thank you for contributing!

Before submitting this pull request, please verify that you have:
 - [ ] Run your code through ESLint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
 - [ ] Run integration tests.
 - [ ] Run end-to-end tests.
 - [ ] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

lomamech committed:
test(Aged Debtors)
Implement test report and expected them with the development-database

lomamech committed:
style(report aged handlebars)
- Use the own style of Bhima's Reports

sfount commented on client/src/js/components/bhReportPeriodSelect.js:
```diff
@@ -0,0 +1,55 @@
+    Fiscal.periodFiscalYear()
+      .then(function (periodes) {
+        periodes.forEach(function (period) {
+          period.monthYear = $filter('date')(period.end_date, 'MMMM yyyy');
```
:+1: This is a very cool solution

sfount commented on server/controllers/finance/fiscal.js:
```diff
@@ -643,3 +644,28 @@ function getPeriodCurrent(date) {
 }
+
+/**
+ * @method getPeriodFiscalYear
```
I think a very similar function was also introduced in @DedrickEnc 's pull request (https://github.com/IMA-WorldHealth/bhima-2.X/pull/1813).

These should eventually be combined to make sure we don't have duplicated code - can you make an issue to track this? 
* Period select component should be combined 
* Fiscal year period utilities on the server should be combined

lomamech replied:
@method getPeriodFiscalYear is somewhat different because the results of this function return the information for the different periods of the fiscal years except for period 0 and 13 and these data are used in the bhReportPeriodSelect component

DedrickEnc replied:
@lomamech 
the method **list** at [fiscalPeriod.js](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/server/controllers/finance/fiscalPeriod.js#L66) can return or not the period 0 and 13. Just pass ```{excludeExtremityPeriod : true or false}``` as parameter. if true the period 0 and 13 will not be returned. [here is the service at the client side](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/js/services/FiscalPeriodService.js).

sfount reviewed (commented):
Code LGTM. I will pull this to review the functionality!

Once @mbayopanda 's balance pull request (https://github.com/IMA-WorldHealth/bhima-2.X/pull/1816#pullrequestreview-48364393) has been merged it would be good to update this to the latest master, it should then be able to be merged.

lomamech committed:
fix(Aged Debtor): resolve conflict with branch Master

lomamech committed:
fix(Conflict with the branch Master)

lomamech committed:
perf(Aged debtor)
- Remove api getPeriodFiscalYear and using fiscal service
- Fix bug in implementation of FiscalPeriodService
- Update api fiscalPeriod.js

lomamech committed:
fix(bhFiscalPeriodSelect): Fix the name of service for prevent error
